cmake/WarpXFunctions.cmake

```
@@ -113,6 +113,19 @@ macro(set_cxx_warnings)
     endif ()
 endmacro()
 
+# Enables interprocedural optimization for a list of targets
+#
+macro(enable_IPO all_targets_list)
```

In CMake, macros have no scope and functions do.

Here, we can go nicely encapsulated with a scope:
```suggestion
function(enable_IPO all_targets_list)
```